shell/platform/linux/fl_settings_portal.cc

```diff
@@ -0,0 +1,263 @@
+  fl_settings_emit_changed(FL_SETTINGS(self));
+}
+
+static FlClockFormat fl_settings_portal_get_clock_format(FlSettings* settings) {
+  FlSettingsPortal* self = FL_SETTINGS_PORTAL(settings);
+
+  FlClockFormat clock_format = FL_CLOCK_FORMAT_24H;
+
+  g_autoptr(GVariant) value = nullptr;
+  if (fl_settings_portal_get_value(self, &kClockFormat, &value)) {
+    const gchar* clock_format_str = g_variant_get_string(value, nullptr);
+    if (g_strcmp0(clock_format_str, kClockFormat12Hour) == 0) {
+      clock_format = FL_CLOCK_FORMAT_12H;
+    }
+  }
+
+  return clock_format;
+}
+
+static FlColorScheme fl_settings_portal_get_color_scheme(FlSettings* settings) {
+  FlSettingsPortal* self = FL_SETTINGS_PORTAL(settings);
+
+  FlColorScheme color_scheme = FL_COLOR_SCHEME_LIGHT;
+
+  g_autoptr(GVariant) value = nullptr;
+  if (fl_settings_portal_get_value(self, &kColorScheme, &value)) {
+    if (g_variant_get_uint32(value) == PREFER_DARK) {
+      color_scheme = FL_COLOR_SCHEME_DARK;
+    }
+  } else if (fl_settings_portal_get_value(self, &kGtkTheme, &value)) {
+    const gchar* gtk_theme_str = g_variant_get_string(value, nullptr);
+    if (g_str_has_suffix(gtk_theme_str, kGtkThemeDarkSuffix)) {
+      color_scheme = FL_COLOR_SCHEME_DARK;
+    }
+  }
+
+  return color_scheme;
+}
+
+static gdouble fl_settings_portal_get_text_scaling_factor(
+    FlSettings* settings) {
+  FlSettingsPortal* self = FL_SETTINGS_PORTAL(settings);
+
+  gdouble scaling_factor = 1.0;
+
+  g_autoptr(GVariant) value = nullptr;
+  if (fl_settings_portal_get_value(self, &kTextScalingFactor, &value)) {
+    scaling_factor = g_variant_get_double(value);
+  }
+
+  return scaling_factor;
+}
+
+static void fl_settings_portal_dispose(GObject* object) {
+  FlSettingsPortal* self = FL_SETTINGS_PORTAL(object);
+
+  g_clear_object(&self->dbus_proxy);
+  g_clear_pointer(&self->values, g_variant_dict_unref);
+
+  G_OBJECT_CLASS(fl_settings_portal_parent_class)->dispose(object);
+}
+
+static void fl_settings_portal_class_init(FlSettingsPortalClass* klass) {
+  GObjectClass* object_class = G_OBJECT_CLASS(klass);
+  object_class->dispose = fl_settings_portal_dispose;
+}
+
+static void fl_settings_portal_iface_init(FlSettingsInterface* iface) {
+  iface->get_clock_format = fl_settings_portal_get_clock_format;
+  iface->get_color_scheme = fl_settings_portal_get_color_scheme;
+  iface->get_text_scaling_factor = fl_settings_portal_get_text_scaling_factor;
+}
+
+static void fl_settings_portal_init(FlSettingsPortal* self) {}
+
+FlSettingsPortal* fl_settings_portal_new(GVariantDict* values) {
+  FlSettingsPortal* portal =
+      FL_SETTINGS_PORTAL(g_object_new(fl_settings_portal_get_type(), nullptr));
+  portal->values =
+      values ? g_variant_dict_ref(values) : g_variant_dict_new(nullptr);
+  return portal;
+}
+
+// Based on
+// https://gitlab.gnome.org/GNOME/Initiatives/-/wikis/Dark-Style-Preference#other
+static gboolean settings_portal_read(GDBusProxy* proxy,
```

Comment:
The (undocumented) convention in the other files is to move the static functions to the top of the file.